core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -523,6 +530,64 @@ public boolean isValidationRequired() {
     return isExtraReadEnabled();
   }
 
+  @VisibleForTesting
+  boolean isMatched(LikeExpression likeExpression, String value) {
+    String escape = likeExpression.getEscape();
+    String regexPattern =
+        convertRegexPatternFrom(
+            likeExpression.getTextValue(), escape.isEmpty() ? null : escape.charAt(0));
+    if (likeExpression.getOperator().equals(Operator.LIKE)) {
+      return value != null && Pattern.compile(regexPattern).matcher(value).matches();
+    } else {
+      return value != null && !Pattern.compile(regexPattern).matcher(value).matches();
+    }
+  }
+
+  /**
+   * Convert SQL 'like' pattern to a Java regular expression. Underscores (_) are converted to '.'
+   * and percent signs (%) are converted to '.*', other characters are quoted literally. If an
+   * escape character specified, escaping is done for '_', '%', and the escape character itself.
+   * Although we validate the pattern when constructing {@code LikeExpression}, we will check it
+   * just in case and throw {@code IllegalArgumentException} for the invalid pattern. This method is
+   * implemented referencing the following Spark SQL implementation.
+   * https://github.com/apache/spark/blob/a8eadebd686caa110c4077f4199d11e797146dc5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala
+   *
+   * @param likePattern a SQL LIKE pattern to convert
+   * @param escape an escape character.
+   * @return the equivalent Java regular expression of the given pattern
+   */
+  private String convertRegexPatternFrom(String likePattern, @Nullable Character escape) {
+    if (likePattern == null) {
+      throw new IllegalArgumentException("LIKE pattern must not be null");
+    }
```

Comment:
Since the validation for likePattern is already done here, I think using assertions is more appropriate:

```suggestion
    assert likePattern != null : "LIKE pattern must not be null";
```